cristin-project/src/main/java/no/unit/nva/cristin/projects/model/cristin/CristinProjectBuilder.java

```diff
@@ -215,4 +218,17 @@ private static CristinExternalSource toCristinExternalSource(ExternalSource exte
         return new CristinExternalSource(externalSource.getName(),
                                          externalSource.getIdentifier());
     }
+
+    /**
+     * Extracts funding source code from URI giving a valid Cristin source code.
+     */
+    public static String extractFundingSourceCode(URI source) {
+        if (isNull(source)) {
+            return null;
+        }
+        var sourceAsText = source.toString();
```

Comment:
You can also just wrap the URI in UriWrapper get call getLastPathElement().

Reply:
I was thinking about http entities like slash, but if it is already an url it might already be urlencoded

Reply:
This supports both uri format and code in string format